src/server/components/lists/find/handlers/translators-interpreters/languages.ts

```diff
@@ -0,0 +1,87 @@
+import type { Request, Response } from "express";
+import { languages } from "server/services/metadata";
+
+import { sanitiseLanguages } from "./../../helpers/sanitiseLanguages";
+import { getLanguagesRows } from "server/models/listItem/providers/helpers";
+import querystring from "querystring";
+
+const languagesAsGovukOptionsArray = splitLanguagesIntoOptionsArray();
+
+export function get(req: Request, res: Response) {
+  const shouldReturn = req.query.return === "results";
+  const queryString = querystring.encode({ ...(shouldReturn && { return: "results" }) });
+
+  if (req.query.remove) {
+    const { languages: sessionLanguages = [] } = req.session.answers!;
+    const languagesAfterRemoving = sessionLanguages.filter((language) => language !== req.query.remove);
+
+    req.session.answers!.languages = languagesAfterRemoving;
+    req.session.answers!.languagesReadable = languagesAfterRemoving.map((lang) => languages[lang]);
+    res.redirect(`languages?${queryString}`);
+
+    return;
+  }
+  const languagesViewModel = languagesAsGovukOptionsArray;
+  const sanitisedLanguages = sanitiseLanguages(req.session.answers!.languages);
+
+  res.render("lists/find/translators-interpreters/languages.njk", {
+    languages: [{ text: "", value: "" }, ...languagesViewModel],
+    languagesRows: getLanguagesRows(sanitisedLanguages),
+  });
+}
+
+export function post(req: Request, res: Response) {
+  let selectedLanguages = sanitiseLanguages(req.session.answers!.languages);
+  const { language, action } = req.body;
+  const shouldReturn = req.session.answers?.disclaimer;
+  const queryString = querystring.encode({ ...(shouldReturn && { return: "results" }) });
+  if (action === "add") {
+    selectedLanguages.push(language);
+    selectedLanguages = sanitiseLanguages(selectedLanguages);
+
+    if (selectedLanguages.length === 0) {
+      req.flash("error", "You must enter the language(s) you need translating or interpreting");
+    }
+
+    req.session.answers!.languages = selectedLanguages;
+    req.session.answers!.languagesReadable = selectedLanguages.map((lang) => languages[lang]);
+    res.redirect(`languages?${queryString}`);
+    return;
+  }
+
+  if (shouldReturn) {
```

Comment:
shouldn't this be a conditional to return to the languages to the results page if req.query.return === "results" like the other post handlers?  I can't see a similar pattern for languages.

Reply:
No. The current behaviour forces the user to go to the next page (page with just the summary of languages).

the languages summary is the last page before the disclaimer. Clicking continue on this page should ordinarily take you to the disclaimer page. The disclaimer handler automatically redirects to the results page if the disclaimer has already been answered.

Reply:
gotcha.  I guess without seeing the full flow it's not so obvious.  Would it be worth specifying the page flows for each service in the README to give the dev some context when starting the project?